llvm/include/llvm/Passes/PassPlugin.h

```diff
@@ -107,7 +107,13 @@ class PassPlugin {
 ///   };
 /// }
 /// ```
+
+#if defined(_WIN32) && defined(LLVM_BUILD_LLVM_DYLIB)
+extern "C" ::llvm::PassPluginLibraryInfo __declspec(dllexport)
+llvmGetPassPluginInfo();
+#else
 extern "C" ::llvm::PassPluginLibraryInfo LLVM_ATTRIBUTE_WEAK
 llvmGetPassPluginInfo();
+#endif
```

Comment:
I think that we can share most of the declaration between the platforms, it is only the attribute that differs.

```c
extern "C" ::llvm::PassPluginLibraryInfo
#if defined(_WIN32)
#if defined(LLVM_BUILD_LLVM_DYLIB)
__declspec(dllexport)
#endif
#else
LLVM_ATTRIBUTE_WEAK
#endif
llvmGetPassPluginInfo();
```

Reply:
That looks a really painful way to declare it.